index.bs

```diff
@@ -488,13 +490,238 @@ partial interface NeuralNetworkContext {
+    <pre highlight="js">
+    let hidden = initialHidden;
+    let slots = (direction == "both" ? 2 : 1);
+    let sequence = null;
+
+    let cellWeight = [];
+    let cellRecurrentWeight = [];
+    let cellBias = [];
+    let cellRecurrentBias = [];
+
+    for (let slot = 0; slot < slots; ++slot) {
+      cellWeight.push(nn.squeeze(nn.slice(weight, [slot, 0, 0], [slot + 1, -1, -1])), [0]);
+      cellRecurrentWeight.push(nn.squeeze(nn.slice(recurrentWeight, [slot, 0, 0], [slot + 1, -1, -1])), [0]);
+      cellBias.push(bias ? (nn.squeeze(nn.slice(bias, [slot, 0], [slot + 1, -1]), [0])) : null);
+      cellRecurrentBias.push(recurrentBias ? (nn.squeeze(nn.slice(recurrentBias, [slot, 0], [slot + 1, -1]), [0])) : null);
+    }
+
+    for (let step = 0; step < steps; ++step) {
+      let cellHidden = [];
+      let cellOutput = null;
+
+      for (let slot = 0; slot < slots; ++slot) {
+        cellHidden.push(nn.squeeze(nn.slice(hidden, [slot, 0, 0], [slot + 1, -1, -1]), [0]));
+      }
+
+      for (let slot = 0; slot < slots; ++slot) {
+        let slice = (slot == 1 || direction == "backward" ? steps - step - 1 : step);
+        let cellInput = nn.squeeze(nn.slice(input, [slice, 0, 0], [slice + 1, -1, -1]), [0]);
+
+        let result = nn.gruCell(
+          hiddenSize, cellInput, cellWeight[slot], cellRecurrentWeight[slot],
+          cellHidden[slot], cellBias[slot], cellRecurrentBias[slot], 
+          resetAfter, layout, activations);
+
+        cellOutput = (cellOutput ? nn.concat([cellOutput, result], 0) : result);
+      }
+
+      hidden = cellOutput;
+
+      if (returnSequence) {
+        cellOutput = nn.reshape(cellOutput, [1, slots, -1, hiddenSize]);
+        sequence = (sequence ? nn.concat([sequence, cellOutput], 0) : cellOutput);
+      }
+    }
+
+    return (sequence ? [hidden, sequence] : [hidden]);
+    </pre>
+    </div>
+</div>
+
+### gruCell ### {#api-neuralnetworkcontext-grucell}
+A single time step of the Gated Recurrent Unit (GRU) recurrent network using an update gate and a reset gate to compute the hidden state that rolls into the output across the temporal sequence of a recurrent network.
+<script type=idl>
+partial interface NeuralNetworkContext {
+  Operand gruCell(long hiddenSize, Operand input, Operand weight, 
+                  Operand recurrentWeight, Operand hidden,
+                  optional Operand bias, optional Operand recurrentBias,
+                  optional boolean resetAfter = true,
+                  optional RecurrentNetworkWeightLayout layout = "zrn",
+                  optional sequence<RecurrentNetworkActivation> activations);
+};
+</script>
+<div algorithm=grucell>
+    **Arguments:**
+        - *hiddenSize*: a {{long}} scalar. The value of the second dimension of the output tensor shape. It indicates the number of features in the hidden state.
```

Comment:
As other methods, it might be better to move this (attribute) argument after the last `Operand` arguments (`recurrentBias`).

Reply:
It makes sense. Thanks for the change.